### PR TITLE
Replace general/osu! ranking criteria glossaries with stubs and links

### PR DESCRIPTION
**Pull request by pishifat**
because people don't really read the ranking criteria like a continuous article, i linked any relevant terms the first appearance in a rule/guideline rather than the the first appearance on the page. this might be overkill so feedback would be good

requires #3450 for sv and str stubs

### CODE REVIEW

File: wiki/Beatmap_Editor/Song_Setup/Primary_metadata_source/en.md [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+  - source
+---
+
+# Primary metadata source
+
+An **primary metadata source** is a source for `Artist`, `Title`, and `Source` fields in a [beatmapset's metadata](/wiki/Beatmap_Editor/Song_Setup#song-and-map-metadata) officially promoted by the artist or their label.
+
+An artist's website, their label's website, CD scans, or official uploads to websites like Bandcamp, YouTube, or SoundCloud are considered primary sources for song metadata. Third party websites such as wikis, databases (vndb, vgmdb, etc.), or music services (Spotify, iTunes, Amazon, etc.) are unreliable and therefore not used for song metadata.
+
+Beatmaps are required to use metadata from any primary source they choose, though the [Ranking criteria](/wiki/Ranking_criteria) prioritizes consistency with other Ranked beatmapsets if their metadata matches another primary metadata source.
```
Comment (cl8n) [inline wiki/Beatmap_Editor/Song_Setup/Primary_metadata_source/en.md:14]: beatmaps ain't required to do anything

File: wiki/Beatmap_Editor/Song_Setup/Primary_metadata_source/en.md [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+  - metadata
+  - song metadata
+  - source
+---
+
+# Primary metadata source
+
+An **primary metadata source** is a source for `Artist`, `Title`, and `Source` fields in a [beatmapset's metadata](/wiki/Beatmap_Editor/Song_Setup#song-and-map-metadata) officially promoted by the artist or their label.
+
+An artist's website, their label's website, CD scans, or official uploads to websites like Bandcamp, YouTube, or SoundCloud are considered primary sources for song metadata. Third party websites such as wikis, databases (vndb, vgmdb, etc.), or music services (Spotify, iTunes, Amazon, etc.) are unreliable and therefore not used for song metadata.
```
Comment (cl8n) [inline wiki/Beatmap_Editor/Song_Setup/Primary_metadata_source/en.md:12]: apparently needs a note about FA listing https://osu.ppy.sh/community/forums/topics/1051871

Comment (cl8n) [inline wiki/Beatmap_Editor/Song_Setup/Primary_metadata_source/en.md:12]: > Third party websites [...] are unreliable

sounds kinda harsh (maybe by accident), the common wikis and databases around just don't follow the same rules that osu! does with metadata.
```suggestion
An artist's website, their label's website, CD scans, and official uploads to websites like Bandcamp, YouTube, and SoundCloud are considered primary sources for song metadata. Third party websites such as wikis, databases (vndb, vgmdb, etc.), and music services (Spotify, iTunes, Amazon, etc.) do not always reflect the artist's intention and should not be treated as primary sources.
```

File: wiki/Beatmaps/Patterns/en.md [hunk 3]
```diff
@@ -0,0 +1,5 @@
+# Gameplay
```
Comment (cl8n) [inline wiki/Beatmaps/Patterns/en.md:1]: wat

File: wiki/Beatmaps/Patterns/en.md [hunk 4]
```diff
@@ -0,0 +1,5 @@
...
+
+- [Jump](/wiki/Beatmaps/Patterns/Jump)
```
Comment (cl8n) [inline wiki/Beatmaps/Patterns/en.md:3]: u can shorten the link cuz its directly under
```suggestion
- [Jump](Jump)
```

File: wiki/Beatmaps/Patterns/en.md [hunk 5]
```diff
@@ -0,0 +1,5 @@
...
+
```
Comment (cl8n) [inline wiki/Beatmaps/Patterns/en.md:4]: ok and where is stream

File: wiki/Beatmaps/Patterns/Stream/en.md [hunk 6]
```diff
@@ -0,0 +1,12 @@
+---
+tags:
+  - streams
+---
+
+# Stream
+
+A **stream** is an [osu!](/wiki/Game_Modes/osu!) gameplay element where consecutive circles are grouped together and intended to be played by alternating clicking between fingers. Streams are usually played with smooth movement, as opposed to a snappy movement of [jumps](/wiki/Beatmaps/Patterns/Jump).
+
+In most beatmaps, streams represent high note density in a song and are snapped to 1/4 ticks on the [hitobjects timeline](/wiki/Beatmap_Editor/Timelines#hit-objects).
```
Comment (cl8n) [inline wiki/Beatmaps/Patterns/Stream/en.md:10]: shouldnt it be like "Most streams represent ..." not most beatmaps
same for jump artcile

File: wiki/Mapping_Techniques/Time_distance_equality/en.md [hunk 7]
```diff
@@ -0,0 +1,13 @@
...
+tags:
+  - time-distance equality
+  - distance snap
+---
+
+# Time-distance equality
+
+**Time-distance equality** is an [osu!](/wiki/Game_Modes/osu!) mapping technique where spacing between [hit objects](/wiki/Hit_Objects) on the playfield is proportional to the duration of time between them. This is automatically calculated by the [distance snap](/wiki/Beatmap_Editor/Distance_Snap) feature of the editor.
+
+Lower difficulties use time-distance equality to make gameplay more intuitive to new players.
```
Comment (cl8n) [inline wiki/Mapping_Techniques/Time_distance_equality/en.md:11]:
```suggestion
Lower difficulties use time-distance equality to make gameplay more intuitive for new players.
```

File: wiki/Ranking_Criteria/en.md [hunk 8]
```diff
@@ -33,104 +33,80 @@ For the record, these include the recent changes made [from the discussion forum
...
+- **Difficulty names in a beatmapset must be clearly progressive and accurately indicating of their respective difficulties, excluding:**
   - The highest difficulty.
   - The highest difficulties with a similar level of difficulty, applying only to Insane and Extra difficulties (e.g. the Insane difficulties of a ENHIIII set or the Extra difficulties of a ENHIIXXX set).
+- **All game modes within a beatmapset must form a spread starting from the lowest difficulty level dictated by the song's [drain time](/wiki/Gameplay/Drain_time),** excluding single-difficulty beatmapsets. Each difficulty must comply with its respective mode's difficulty-specific Ranking Criteria, the spread cannot skip any difficulty levels, and there cannot be any drastically large difficulty gaps between any two difficulties.
```
Comment (cl8n) [inline wiki/Ranking_Criteria/en.md:49]: > the spread cannot skip any difficulty levels, and there cannot be any drastically large difficulty gaps between any two difficulties.

this is a new part of the rule, dont change rules in this pr

Comment (cl8n) [inline wiki/Ranking_Criteria/en.md:49]: "excluding single-difficulty beatmapsets" is useless, a single diff set at appropriate difficulty does comply with the spread rule

File: wiki/Ranking_Criteria/en.md [hunk 9]
```diff
@@ -33,104 +33,80 @@ For the record, these include the recent changes made [from the discussion forum
...
-- **Official Sources must be used as references for metadata unless none are available.** In that case use what is most common and recognizable.
-- **Do not modify the metadata an artist provides on official sources unless said modification is done in order to comply with formatting and standardisation rules on this Ranking Criteria.**
+- **[Guest difficulty](/wiki/Beatmaps/Beatmapsets/Guest_difficulty) creators, storyboarders, skinners and hitsounders must be added to the tags of a beatmapset.** This is to give credit where credit is due and help others identify the main contributors of any given beatmapset. Usernames containing single characters separated by spaces must have the spaces replaced with underscores.
+- **[Official metadata sources](/wiki/Beatmap_Editor/Song_Setup/Primary_metadata_source) must be used as references for metadata.** Do not modify metadata from official sources except to comply with formatting and standardisation rules below. If no sources are available, use what is most common and recognizable.
```
Comment (cl8n) [inline wiki/Ranking_Criteria/en.md:72]: why not call it primary metadata source here too

File: wiki/Ranking_Criteria/en.md [hunk 10]
```diff
@@ -33,104 +33,80 @@ For the record, these include the recent changes made [from the discussion forum
...
+- **If a symbol is used to group parts of a title, a space must be used before and after the group, but not directly before or after the symbols within the groups.**
 - **If a series applicable as a source contains sub-series, the most precise label must be used as the source.** If a song has multiple applicable sub-series, the main series/franchise may be used instead.
 
 #### Romanisation
 
 - **Artist names are to be romanised in the order they are printed in the unicode fields.**
 - **Loan words from other languages have to use the original words in their stead when attempting to romanise them.**
 - **When a song uses repeat words in the title or in the artist where one is in unicode, and the other as a basic romanisation, the romanised field must use the provided romanisation only and remove the duplicate word.**
-- **Songs with German metadata must romanise umlauts into two-letter equivalents (`ue`, `oe`, `ae` and `ss`).**
+- **Songs with German metadata must romanise umlauts into two-letter equivalents: `ü` to `ue`, `ö` to `oe`, `ä` to `ae` and `ß` to `ss`.**
```
Comment (cl8n) [inline wiki/Ranking_Criteria/en.md:102]: not only a german thing.
```suggestion
- **Umlauts must be romanised into two-letter equivalents: `ü` to `ue`, `ö` to `oe`, `ä` to `ae` and `ß` to `ss`.**
```

File: wiki/Ranking_Criteria/en.md [hunk 11]
```diff
@@ -33,104 +33,80 @@ For the record, these include the recent changes made [from the discussion forum
...
+- **Songs with Japanese metadata must use the [Modified Hepburn romanisation](https://en.wikipedia.org/wiki/Hepburn_romanization#Features) method in romanised fields.** The same applies to the Source field if a romanised Source is preferred by the mapper. As a non-unicode field, long vowels such as `おう` and `うう` should be romanised into `ou` and `uu` to avoid macrons.
+- **Songs with Chinese metadata are to be romanised with respect to the tones and dialects of Chinese they belong to.** Each Chinese character must be romanised as a capitalised word and separated with a space, with the exception of artist names. In any case, all diacritical tone marks must be omitted:
   - Mandarin metadata must be romanised using the Hanyu Pinyin system.
   - Cantonese metadata must be romanised by using the Jyutping system.
   - If the song falls into neither category, this choice is left up to the mapper's discretion and contacting a native speaker is recommended.
-
+Each Chinese character must be romanised as a capitalised word and separated with a space.
```
Comment (cl8n) [inline wiki/Ranking_Criteria/en.md:109]: no
```suggestion

```

File: wiki/Beatmaps/Beatmapsets/en.md [hunk 12]
```diff
@@ -1,10 +1,16 @@
...
+  - beatmap
+  - mapset
+---
+
 # Beatmapsets
 
 <!-- TODO: connect to the rest of the wiki, ultimately this should replace the Glossary entry and be clearly distinguished from "beatmaps" -->
 
-![An in-game screenshot of a beatmapset in the song selection menu](img/in-game.jpg "Sing's beatmapset of \"nanobii - popsicle beach\"")
+![An in-game screenshot of a beatmapset in the song selection menu](img/in-game.jpg "Beatmapset of \"nanobii - popsicle beach\" hosted by user \"Sing\"")
```
Comment (cl8n) [inline wiki/Beatmaps/Beatmapsets/en.md:11]: why change tho this sounds worse

File: wiki/Beatmaps/Beatmapsets/en.md [hunk 13]
```diff
@@ -1,10 +1,16 @@
...
 
-A **beatmapset** is a set of [beatmaps](/wiki/Beatmaps) for the same song. The [beatmap listing](https://osu.ppy.sh/beatmapsets) shows all of the beatmapsets submitted online.
+A **beatmapset** is a set of [beatmaps](/wiki/Beatmaps) for the same song managed by a [beatmapset host](/wiki/Beatmaps/Beatmapsets/Beatmapset_host). The [beatmap listing](https://osu.ppy.sh/beatmapsets) shows all of the beatmapsets submitted online.
```
Comment (cl8n) [inline wiki/Beatmaps/Beatmapsets/en.md:13]: too specific to the common pattern of host+guests on submitted beatmaps imo, a beatmapset has nothing to do with this by default

the hsot +guest thing could be next paragraph or something

File: wiki/Beatmaps/Beatmapsets/Guest_difficulty/en.md [hunk 14]
```diff
@@ -0,0 +1,10 @@
+---
+tags:
+  - guest difficulties
+---
+
+# Guest difficulty
+
+A **guest difficulty** is a difficulty of a [beatmapset](/wiki/Beatmaps/Beatmapsets) that is not created by the [beatmapset host](/wiki/Beatmaps/Beatmapsets/Beatmapset_host). A guest difficulty featuring multiple mappers is also known as a [collaborative difficulty](/wiki/Beatmaps/Beatmap_collaborations)
```
Comment (cl8n) [inline wiki/Beatmaps/Beatmapsets/Guest_difficulty/en.md:8]: > A guest difficulty featuring multiple mappers is also known as a collaborative difficulty

technically true i guess but the host could also participate in collab which makes this sentence confusing. it sounds like ur talking about strictly multiple guests

File: wiki/Beatmap_Editor/Song_Setup/Primary_metadata_source/en.md (hunk 2 above)
Comment (pishifat) [reply wiki/Beatmap_Editor/Song_Setup/Primary_metadata_source/en.md:12]: shouldn't need this

File: wiki/Ranking_Criteria/en.md (hunk 8 above)
Comment (pishifat) [reply wiki/Ranking_Criteria/en.md:49]: this is what "reasonable spread" was defined as in the original rule, so it isn't changed?

**Comment by cl8n** [comment]
the edits to Beatmap_Editor/Timing don't really get across that this is supposed to be describing the Timing tab of the editor, not Timing as a concept

for now I'm gonna try to combine what you removed and added but you may want to add another article Beatmapping/Timing or somethn to link when you're talking about the concept

**Pull request merged**